packages/signature-controller/src/SignatureController.ts

```diff
@@ -276,11 +283,13 @@ export class SignatureController extends BaseController<
    *
    * @param messageParams - The params of the message to sign & return to the Dapp.
    * @param req - The original request, containing the origin.
+   * @param traceContext - The parent context for any new traces.
    * @returns Promise resolving to the raw data of the signature request.
    */
   async newUnsignedPersonalMessage(
     messageParams: PersonalMessageParams,
     req: OriginalRequest,
+    traceContext?: TraceContext,
```

Comment:
As this is a specialist argument, and we want it future-proof, should we include a generic options object like we do with `addTransaction`?

Reply:
Done in https://github.com/MetaMask/core/pull/4655/commits/4389c98462520921f1bed7afd2a351ed02965164